### ISSUE TITLE
Constant connection drop Harman discover pro + Samsung s22 ultra

### ISSUE THREAD

**Issue by worldonfire** [cpebit/aawireless-issues#124]
**Describe the bug**
There are constant random connection drops when driving (even when being standstill.). There are no certain places, where connection is lost. Sometimes it drops 1-2 times per 20minute drive, sometimes it drops more than 10 times.
LED light on device - at first turns off. Then solid green for some time, then blinks green and then blue.
Have tried different device settings - dongle mode, passthrough, different usb modes and so on.
With dongle mode + passthrough enabled, it went to bootloop - at first worked for some 3-5 minutes and then loop started. When passtrhough disabled, it just restarts randomly.
Have tried 3-5 different usb cables, even bought one brand new cable (also have tried few aawireless devices).
At the moment having 3 aawireless devices. Second one is tested in another passat with same headunit + Oneplus 6 - also random connection drops.
Third one is in 2017 Tiguan with Technisat headunit and Samsung Note 10 - also random disconnections.

**Device name (AAWireless/AndroidAuto-AAWxxxxxxx) (please complete the following information):**
AndroidAuto-AAWQin7o2w1

**Car (please complete the following information):**
Vw Passat B8 2017

**Headunit (please complete the following information):**
Discover Pro 9.2" Harman with latest SW

**Smartphone (please complete the following information):**
 - Samsung Galaxy S22 Ultra
 - Android 12
 
 **Country:**
 Estonia

**Additional context**
One bug report reported already through companion app (passtrough bootloop problem)

**Comment by Snirpo** [comment]
Very very odd to have this thing happening with three different devices. Sounds like the device fully turns off. Do these devices all have firmware 2.0? Can you pass the device names of all devices?

**Comment by worldonfire** [comment]
As i'm not driving with other cars all the time, I can only say as much as other users said to me.

**Tiguan 2017 with composition media (technisat) and samsung galaxy note 10:**
Leds act same as in my car: off, solid green, blinking green and blue. This user has tried different settings, different wifi channels etc, still random reboots.
current sw version is 2.0 and device name: AAWireless-l7zPPeZI  (this is the replacement device i got. If I remember correctly, manufactured somewhere beginning of 2021?). Used this one in my car also with different usb cables. At first it was propably 1.12 firmware. Tried different settings but no luck and then upgraded to 2.0.. And then gave it to my friend as other two devices arrived.

**Other passat with harman discover pro and oneplus 6:**
This one is less tested as the person using it does not have so much patience as I have :)
One drive when I set the device up, it did quite many restarts. Did set it to passtrough, startstop and mtp and it remained quite stable to the end of the drive, but next time it was used, it started the disconnection again. It was tested few times again, but unfortunately I cannot confirm, what the leds did.
Yesterday took that oneplus 6 and went to see versions. At first it was 2.0fw but the connection did not want to stay on (tried different settings. In the end if I remember correctly, led was constantly blinking blue and on headunit show that select Android auto or Mirrorlink, but instantly disconnected and then again and so on... like the initializing was looping constantly. Then did software udpate to 2.0.1 but it remained the same. No matter if I changed settings (passtrough, dongle mode, usb mode etc...)
Device id before update: AndroidAuto-AAW769eHRGM
device id after update: AndroidAuto-AAWcT3Xy8kG

Device that i'm using currently in 2017 passat with discover pro harman:
Id: AndroidAuto-AAWQin7o2w1 and 2.0.1 software.


In my car and in other passat, all the usb cables that have been used, have worked perfectly when using AndroidAuto wired.

**Comment by Snirpo** [comment]
Ok, let's first focus on your own car. To confirm, these things also happened on pre-2.0 firmwares?

**Comment by Snirpo** [comment]
Ok, small update. I did a bit of research in Freshdesk, and found quite some tickets with the same issue. From 8 months ago even.

For our own reference:
https://cpeb.freshdesk.com/a/search/tickets?term=passat%20b8

**Comment by worldonfire** [comment]
I cannot confirm 100% what the lights did with pre 2.0 firmware, but yes, disconnection have happened with every firmware. 
I'm not quite sure if it was 1.8 or 1.12, but one of those was bit more stable - usually in the mornings managed to do the 15-20min drive with zero or one disconnection. (passtrough, mtp and startstop enabled). Just remembered that current device has been from the beginning 2.0 or higher sw.

If I'm able to help to debug, i'll help as much as I can :)

**Comment by Sasha0088** [comment]
Good afternoon, the same situation as the author. After 2-10 minutes, it reboots itself, first green, then blinking green, then blue. Samsung S20 phone. Passat B8 2016. Harman - AAW1Zv8A7i1. I do the same thing, I changed the bands 5/2.4, the frequency. Cable.

**Comment by Snirpo** [comment]
@Sasha0088 or @worldonfire do you live in the EU by any chance. I want to test if these issues are fixed by using a separate power source from cigarette power adapter. I can send you the hardware to do this.

**Comment by Snirpo** [comment]
@Sasha0088 Can you send me a PM on https://join.slack.com/t/aawireless/shared_invite/zt-190iap90x-VNUIRNgbvAjHMDcrlqHJeg . Let's see if we can arrange something. I'm pretty sure these issues are hardware related.

**Comment by worldonfire** [comment]
> @Sasha0088 or @worldonfire do you live in the EU by any chance? I want to test if these issues are fixed by using a separate power source from cigarette power adapter. I can send you the hardware to do this.

I'm in EU - Estonia

**Comment by Snirpo** [comment]
@worldonfire great! Can you send me your address details on Slack? https://join.slack.com/t/aawireless/shared_invite/zt-190iap90x-VNUIRNgbvAjHMDcrlqHJeg

**Comment by Sasha0088** [comment]
nickname Sasha0088


ср, 8 июн. 2022 г., 15:53 Chiel Prins ***@***.***>:

**Comment by worldonfire** [comment]
> @worldonfire great! Can you send me your address details on Slack? https://join.slack.com/t/aawireless/shared_invite/zt-190iap90x-VNUIRNgbvAjHMDcrlqHJeg
> 
> PM on XDA is also fine.

Tried to send message on XDA.

**Comment by Snirpo** [comment]
UPDATE:

I'll be sending Y splitter cable + siggarette adapter to @worldonfire and @Sasha0088 . Because the device is fully rebooting, I have the feeling this problem is USB power related.

**Comment by jaszcz0mp** [comment]
Hi, I have a similar problem. My equipment: VW Passat b8, mib2 std technisat, MTS_EU_VW_ZR_P0478T
and
GALAXY S22 ultra updated firmware.

Android Auto 7.7.622104-release.

I tried to use the power splitter https://www.amazon.pl/gp/product/B07FJZZ7FK/ref=ppx_yo_dt_b_asin_title_o00_s00?ie=UTF8&psc=1
unfortunately this did not solve the problem. It works best when I use this: https://en.aliexpress.com/item/1005001894983964.html?spm=a2g0o.order_detail.0.0.746b43ceOB8GjJ&gatewayAdapt=glo2pol (Male A to Male c)
unfortunately, reboots happen at random times. Here is a video of what it looks like https://drive.google.com/file/d/1DwxNopIGi0hB3SsCHNxBpyWcaZyAKYly/view?usp=drivesdk

**Comment by Snirpo** [comment]
@jaszcz0mp This video is very helpful, because it does not seem to be an immediate power loss, the LED stays blue for a while. Seems more like a communication loss and the HU then deciding to reset USB power.

**Comment by jaszcz0mp** [comment]
In my car, when connected via cable, AA is very sensitive to the quality of the cable. Works best with original samsung cable

**Comment by Snirpo** [comment]
@jaszcz0mp Can you confirm this issue also existed on firmware 1.5.0?

**Comment by jaszcz0mp** [comment]
unfortunately no, I haven't used AAWireless with this car before 2.0

**Comment by jaszcz0mp** [comment]
unfortunately no, I haven't used AAWireless with this car before firmware
2.0

śr., 15 cze 2022 o 09:01 Chiel Prins ***@***.***> napisał(a):

**Comment by Snirpo** [comment]
@jaszcz0mp Do you mind reverting to 1.12.0 for a test?

**Comment by Sasha0088** [comment]
@Snirpo 
Here's how I reboot
https://drive.google.com/file/d/1KgykiYrhfB8SKEda5NWIn5Jfkrm_Rd14/view?usp=drivesdk

**Comment by worldonfire** [comment]
> @Snirpo Here's how I reboot https://drive.google.com/file/d/1KgykiYrhfB8SKEda5NWIn5Jfkrm_Rd14/view?usp=drivesdk

Can you make the link public? At the moment at least i cannot see it.

**Comment by Sasha0088** [comment]
@worldonfire
And now

**Comment by Snirpo** [comment]
@jaszcz0mp for some reason your device has been set to 1.8.0, do you by any change also have contact with our support? Anyway, it doesn't really matter because 1.8.0 is bassically same as 1.12.0. Can you shoot in a bug report via the app? Please press the "start debug mode" button and when the problem has occured, press "stop debug mode" and send the report.

**Comment by worldonfire** [comment]
> @jaszcz0mp This video is very helpful, because it does not seem to be an immediate power loss, the LED stays blue for a while. Seems more like a communication loss and the HU then deciding to reset USB power. I also see this is a pretty old device. This the same problem exist on firmware 1.5.0?
> 
> @worldonfire or @Sasha0088. Can you confirm the LED acts the same in your cars as in the video?
> 
> Surprisingly a different head unit, but the same car, same problem. I did not expect that.

@Snirpo just checked how the leds act - the moment headunit shows that connection is lost, the same moment led on aawireless goes off. Then after short period, it will be white for a short moment, then solid green, then blinking green, then blinking blue and if it manages to start android auto automatically, then solid blue (sometimes i need to manually select android auto on headunit screen).

**Comment by Snirpo** [comment]
@worldonfire That seems different, like an immediate power loss. But could be the same thing as well, depending on how the HU reacts to power loss.

Unfortunaly we're getting nowhere with this issue :( I'll rent a VW Passat myself to check this issue. Hopefully I can reproduce it and debug this issue with an USB debugger to see what's happening.

**Comment by worldonfire** [comment]
There are quite many different type of headunits in passats:
Appconnect support came with MY16 (discover/composition media 6.5" and discover pro 8" harman).
With MY18 came facelift with glass screens - discover/composition media got 8" and discover pro got 9.2".
Also, some cars have usb hub  - like the car my wife is driving - one usb in front and second one inside armrest.
Looks like this usb hub does not affect how the device acts as it restarts with both usb ports.
In my car, there is only one usb port inside armrest.
Both passats have 9.2" discover pro harman headunits.

If there is a way to help to debug, let me know :)

**Comment by Snirpo** [comment]
@worldonfire I have the Y splitter cable. I can still send you one if you want to test? Someone on XDA tested with a Y splitter and it didn't help, but who knows....

**Comment by worldonfire** [comment]
If possible, then I would test, as I really want to get this device working :)

**Comment by Snirpo** [comment]
Sure! I'll create a package and send it tomorrow.

**Comment by Snirpo** [comment]
Intresting observation:

One of our testers on Slack (82:FC:B1:9F:7F:93), has the following HU where AAW works perfectly fine, I quote:
"Last week I had abt 800 km drive, mixed highway and mountain roads without any disconnections of AA or HU crashes."

HU details:
2017 | TechniSat | MIB Standard2 ZR Nav | Volkswagen |  0359
This is a Discover media

@jaszcz0mp 
2018 | TechniSat | MIB Standard2 ZR Nav | Volkswagen | 0478

Seems to me like the same HU. Only difference is in the HU firmware being newer. So my guess it, at least for the Technisat HU, something is HU firmware has changes which makes it not like AAWireless.

**Comment by xsichtasdf** [comment]
Hey, I've received my AAWireless today and it doesn't work, issue sounds like the one reported. So, works for like 5 sec, then disconnects. I also drive a Passat B8 with firmware listed (0478). 

I'm thinking about updating the car to the newest available firmware what is, based on my knowledge, 0480.

Do you perhaps heard about other users who had the same issues with newest firmware? 
Further, does it make sense to contact AAWireless support so the topic get's the necessary attention?
Br

**Comment by Sasha0088** [comment]
@xsichtasdf 
I think it's not in the firmware that I have 1367 and also the same as everyone else.

**Comment by Snirpo** [comment]
@xsichtasdf This is AAWireless tech support. Please don't contact support won't help. It already has the attention right here :)

**Comment by xsichtasdf** [comment]
@Sasha0088 : I was referring to the update of Snirpo from 17h ago. This is related to TechniSat devices, at least based on the firmware number. Sorry for the confusion.

@Snirpo : it was renamed several times (don't know why). Current name: AAWoj1MlcFz

Give me a hint if I should plug it in again and retry

**Comment by Snirpo** [comment]
@Sasha0088 

I see two cars in the backend linked to your device:

2018 | TechniSat | MIB Standard2 ZR Plus | SKODA |  0478
CLU8_ER | HARMAN | 61852 | Volkswagen | MHI2_ER_VWG13_K4525 | CLU8_MMX2_VW_ER_G13_129PROD

Can you confirm on the Skoda, the same issue exists?

**Comment by Sasha0088** [comment]
@Snirpo 
No, I do not confirm, I try to meet him again and ride with him more. Hope to see you soon :)

**Comment by Snirpo** [comment]
I need to debug this low level bassiccly, attach an USB debugger and see whats happening. The problem is, even when I have the Tiguan on Monday, there's a good chance it will just work fine. But we'll see.

**Comment by Snirpo** [comment]
Just tested on a VW Tiguan, sadly it was a MIB3 HU, which works perfectly fine.

**Comment by xsichtasdf** [comment]
Damn. So, what options are left?

**Comment by Sasha0088** [comment]
@Snirpo 
I traveled for 1 hour on a Skoda, I never rebooted, I don’t know how to check it

**Comment by Sasha0088** [comment]
> @Snirpo I traveled for 1 hour on a Skoda, I never rebooted, I don’t know how to check it
2018 | TechniSat | MIB Standard2 ZR Plus | SKODA |  0478

**Comment by Sasha0088** [comment]
> @Snirpo I traveled for 1 hour on a Skoda, I never rebooted, I don’t know how to check it
2018 | TechniSat | MIB Standard2 ZR Plus | SKODA |  0478

**Comment by Snirpo** [comment]
I'll try to get one of these head units to put on my desk.

**Comment by Snirpo** [comment]
One of our testers has a:

2018 | TechniSat | MIB Standard2 PQ Nav | Volkswagen |  0480

And no issues at all. We could try a replacement device I guess, to rule out any device related issues.

**Comment by Sasha0088** [comment]
@Snirpo 
How can I do that

**Comment by xsichtasdf** [comment]
@Snirpo : that's an other Navi I've got, at least on what I've found via Google regarding the device description of yours. 

Would be interested in another test device as well, especially as the delivered usb cable as faulty. (I'm using my regular usb-c cable with which I'm doing android car regularly).

**Comment by Snirpo** [comment]
@xsichtasdf Sure, can do. Can I have your Indiegogo contribution number? I'll send you a new one.

**Comment by xsichtasdf** [comment]
Send you to your Gmail address 👍

**Comment by Snirpo** [comment]
@xsichtasdf Can you possible try the latest develop version of the firmware? I can't find you device unfortunaly, what's the latest name?

@worldonfire @jaszcz0mp I enabled the develop firmware for your devices. This makes the device (almost) one on one replicate the MA1.

**Comment by worldonfire** [comment]
Hi!
Will do the update soon.

Just one interesting thing - my friend wanted to make usb cable with extra power (Y cable). Have used this few days. Managed to drive ~180km trip without a single drop. Next day driving to work, it had one drop. Same evening at same spot another drop (looks like found one specific spot, where device discconnects - aa wireless does not restart.)
After that tried to test, what happens if i don't use extra power. For my surprise - nothing. Had two 100km trips without a single drop.
Also - this usb c part sits more tight in AA Wireless usb port. All the cables that i have tested, they work flawlessly directly with phone.
After 5 days of testing, it starts to look like a cable problem still... But will test it more to be sure.
This made me also wonder - how can a device be so selective with cables?

**Comment by Snirpo** [comment]
Hmm that's very good info. What you are talking about seems interference related issues. Setting device to a higher wifi channel should help. If this does not help, we'll replace your device by a new one.

So it seems like a device-side usb port issue? Hmm, that's interesting. Is this cable also shorter by any chance?

About the cable... I honestly don't know. I've had 100's devices in my hands, with numerous HU's. Never had an issue with the standard cable. It happens, also with wired connections. I think that's one if the great mysteries of AA.

It could also be it's a certain batch which is problematic. Because of component shortages it could very well be the USB port manufacturer has changes a few times.

**Comment by Snirpo** [comment]
I wish I had an answer, but I don't. I don't think these problems on the Technisat and Harman units are the same issue. The problem with random disconnections is that is could be anything, and it's not always easy to read from the logs whats happening. We can't see whats happening on either side (HU / phone), we can only see what the device is doing. I cannot really debug this issue without having an actual car and device with this issue, that's the main problem.

The shorter cable length is very intresting. Because from what I can see in the logs, the HU just completely drops the device. Shorter cable means less resistance, should be negligible normally, but I guess when it's already running on the limit, it can just throw it over the limit.

**Comment by worldonfire** [comment]
Yes, can totally understand :)
If there is a way how to help with debuging- let me know :)

Cable length - so the device could have a power deficite/lack of power?

Yes, I'm using AAWQin7o2w1 currently.

Few weeks ago I have also ordered one cable from aliexpress that should work (at least one user reported that is sits in type C port firmly).  Lets see if that one works...

**Comment by worldonfire** [comment]
Will do!

Would really like to show the car, but i think the drive would be quite long :D

**Comment by Snirpo** [comment]
Yeah I think so. I'm curious if it stays stable now with shorter cable. I'll try to get a hold of one of these cars somehow. Need to attach the USB debugger to probably debug it.

**Comment by worldonfire** [comment]
![20220624_204509](https://user-images.githubusercontent.com/732000/175785732-75c91228-fa60-4a99-b9b6-26b534e0cfd2.jpg)
This is the current "hack" cable. Btw, it fits more firmly into type c socket than the original cable.
I'll try to find some more short cables and test with those.

Also will try to test this cable on my wifes passat to see, if it starts working there.

Btw, updated to latest dev fw. But near one big shopping center, still connection loss. Haven't had oportunity to do longer trips still...

I'm not sure if correct place to ask, but start-stop feature - will this be disabled(noticed something about disabling in forum)? MIB headunits/vw cars keep their usb power on for quite some time. And car powers up usb ports even for example when I enable webasto (noticed it when walked my dog ~50-100m away and phone connected to aawireless device :O)
At the moment not using this, but would like to try again, as it worked quite OK. But this constant restarting made HU or AAwireless sometimes acting weird and it did not always connect back.

**Comment by domik82** [comment]
It's happeening with my car VW PASSAT B8 2015 with Discovery Pro head unit. I can collect more details if needed like version etc. 

I don't use my car that often post Covid so maybe this is just my feeling but.. 

I have 2 USB connectors - one with induction charger in place of cigarete lighter and second under arm rest. 

My feeling is that the issue is more severe when device is plugged in in the front. So maybe the interference is higher caused by more electronic in front. 

In the same time that doesn't really explain why this issue is so regular. 

I had observation for 2 days (before I changed config and version) that it's happening every 7 minutes. The car is showing me how long I'm driving and issue was hitting 8 / 15 min mark on 3 different trips. 

I don't know if there is a way to collect device memory usage profile but if there is a mem leak then it should be visible on graph. 

@Snirpo I know you were searching for a way to reproduce it. There are plenty of B8 passat groups on Facebook . We are exchange info there on how to Fix stuff or info about seeing the car. 

Maybe there is one in your hometown - that would greatly increase chances to check right car (not to mention that if this issue wouldn't exist it's chance of promotion of the device).

Thanks.

**Comment by worldonfire** [comment]
@domik82 are you sure you have wireless charger in front? From factory 2015 could have cellural signal amplifier in front.
Wireless charger ofc can be retrofitted.

**Comment by Snirpo** [comment]
@domik82 Kinda surprised it's so regular. Memory leak is not really possible, then all devices in all cars should be effected, and we would have a very big issue :). I've also been monitoring memory leaks regarding another issue (#116) , and there are no issues.

I'm actually on contact now with a company which does VAG diagnostics. They have a Harman HU for sale. Hopefully I can get it somewhere this week.

Logs are of course always welcome, you can create a debug log via help and support --> bug report in the app.

**Comment by xsichtasdf** [comment]
> @xsichtasdf Can you possible try the latest develop version of the firmware? I can't find you device unfortunaly, what's the latest name?
> 
> @worldonfire @jaszcz0mp I enabled the develop firmware for your devices. This makes the device (almost) one on one replicate the MA1.

Sorry, was away. Updated and will test tomorrow morning. Current name: AAWoj1MIcFz

**Comment by domik82** [comment]
> @domik82 are you sure you have wireless charger in front? From factory 2015 could have cellural signal amplifier in front. Wireless charger ofc can be retrofitted.

To be honest - I thought it's charger - looked today and I'm not so sure anymore. None of my phones fit there so can't even check it.

Anyway I had used a suggestion and found really short USB cable (I got ~20 cm Sony one) - did six around 15 min rides - not a single drop. 

Will observe more.

**Comment by worldonfire** [comment]
> To be honest - I thought it's charger - looked today and I'm not so sure anymore. None of my phones fit there so can't even check it.

Yep, it sounds like signal amplifier. :)
If it would be wireless charger, then on right side, there wouldn't be usb port. With newer models, there is type c usb port almost next to gear lever.

**Comment by xsichtasdf** [comment]
@Snirpo : tested the update. Not once I was able to get the connection up and running. While direct connecting my phone via cable, it worked immediately

**Comment by Snirpo** [comment]
This most definitely is some kind of hardware issue. Can't explain it otherwise. I'm waiting for my test HU to come in.

**Comment by Snirpo** [comment]
OK I just figured something out. All the problems with the Harman HU's are happening with the gen2 devices:

92:91:39:3B:6D:05 -- rv1109
92:91:07:65:46:E0 -- rv1109
36:E7:59:B5:E9:D4 -- rv1109
D2:77:6D:36:E9:C5 -- rv1109

So it might very well be related to the gen2 hardware. We tested a return of someone (92:91:07:65:46:E0), and there were no issues at all with another JVC headunit.

This makes me believe there's some kind of low level issue, be it hardware or kernel level software with these HU's combined with the gen2 devices.

**Comment by Snirpo** [comment]
Harman test setup is being sent today, I'll keep you up to date!

**Comment by Snirpo** [comment]
So received the setup. But unfortunately AA will not start up because of the date / time being incorrect on the HU. Cannot change it either, because ignition is nessecary to change the date. Need some extra components for that apparently... Tried to hack in to it to set the time and date directly, but haven't found out how to do it.

**Comment by worldonfire** [comment]
We deal with VAG group cars here in Estonia. Also with MIB HUs (for example Android Auto enabling) :) PM me and lets talk directly what is needed?

**Comment by xsichtasdf** [comment]
@Snirpo : new device arrived, thanks. Seems to work much smoother, but: issue still persists :(. 
Was able to get a connection up and running for exactly a second after which device turned on the green LED again...
After words, blue led is pinging but the device is not available in HU.

Device name: AAWn8XWItky
Any idea based on the device Infos? Tried different connection methods as well, no change of behaviour...

**Comment by Snirpo** [comment]
@xsichtasdf Kinda expected that. I sent @worldonfire one of my golden oldies (batch 3), let's see how that goes. 

@worldonfire It seems I need some extra modules to make the HU think there's ignition. I do have root access to the HU (MMX/RCC) already toyed around with it a bit, but couldn't find a way yet to hack in the date/time or make the HU think there's ignition. I'm in contact with the VAG specialist, and trying to find a solution with him.

**Comment by worldonfire** [comment]
Yup, you should somehow emulate ignition. For example rasperry or smth.
Lets you they can provide you a solution :)

Got dhl message, hopefully on the 5th i'll get the device and give some reports, how it works :)

**Comment by Snirpo** [comment]
I think I'm able to fake ignition. Bought a simple CAN module to try it out.

**Comment by Snirpo** [comment]
Ok, so we can exclude any device model specific issues. That's good I guess.

**Comment by karlmorris** [comment]
Same issue here using this head unit: https://electronics.sony.com/audio/car-marine-audio/receiver-players/p/xavax5000
Mobile device is a 2020 Motorola Edge running Android 11.

**Comment by worldonfire** [comment]
> OK faking ignition works, but unfortunately I can still not set date / time.

It does not let change settings?
What happens when you connect gps antenna(there is possibility to set the clock automatically via gps)

**Comment by Snirpo** [comment]
OK I got the head unit working. There's an "overrideTime" property in the gal.json where you can override the current time.

![WhatsApp Image 2022-07-15 at 4 49 22 PM](https://user-images.githubusercontent.com/4076368/179263577-4ebc89ab-635a-4b42-ab1b-15f97c77da95.jpeg)

I did some limited testing, but I can't reproduce this issue. I guess this is good, because that means not every Harman VAG HU is affected. I'll to some more thorough testing next week.

**Comment by Snirpo** [comment]
Did some more longer runs, up to an hour and it seems to run fine. So at this point it's still a mystery.

**Comment by Ralys18** [comment]
> Did some more longer runs, up to an hour and it seems to run fine. So at this point it's still a mystery.

Hi @Snirpo I believe I have similar issue with passat b8 also, but from what I tested so far the issue seems like only happen while using Spotify + Google map, if I only use the map for navigation without Spotify I won't get any connection drop so far, hope this helps.

**Comment by Ralys18** [comment]
> @Ralys18 what's your device name?

AAWireless-JCN8iTlp

**Comment by Snirpo** [comment]
@Ralys18 This is a Technisat HU, so different issue. Please open another issue for this issue.

**Comment by xsichtasdf** [comment]
@Snirpo : how do we proceed in this case? Any further ideas? If not, is a refund possible?

**Comment by Snirpo** [comment]
@xsichtasdf I have no real solution at this point. I'll ask our factory to research if they can find anything on the hardware side which could cause this issue. My guess is it's power related.

Refunds are possible, you can contact support at support@aawireless.io and request a refund. You can mention this post.

**Comment by worldonfire** [comment]
I would say that with this one specific "hack" cable (propably with random y cable also, have not managed to test it) AAWireless finally works as expected. It is self made Y cable but I don't use secondary power(second usb cable sits unused). Did ~5000km car trip and used waze every day. Had only few drops in big cities (Warsaw, Vienna and one city in Czech) but these were propably due to wrong wifi channel). Otherwise it worked perfectly.

I will try to test with some other Y cables also.

**Comment by Snirpo** [comment]
@worldonfire Good to know, so it's definitely cable related. I'll ask the factory if there's any reasonable explanation for this.

**Comment by worldonfire** [comment]
Yes, somehow... but does not sound logical at all... have tried so many cables with different lengths and only this hacky one works.

If I have possibility to drive with my wifes passat and remember to take this cable with me, will test it there also.

**Comment by xsichtasdf** [comment]
@worldonfire : can you specify which hacky cable did work in the end? Would like to test it as well

**Comment by worldonfire** [comment]
> 

Please look this post: https://github.com/cpebit/aawireless-issues/issues/124#issuecomment-1166339175
It is self made Y cable. It is ~10cm long and has extra type A(maybe half a meter long) to use with cicarette lighter. BUT by some reason, this cable works without this extra power.

Have tried several other short cables, but with everyone connection drops.
My friend will make another Y cable in near future and i have ordered few of these from aliexpress also.
Will let you know, how with those other cables it will work.

**Comment by Snirpo** [comment]
@worldonfire I'll send you an old style AAWireless cable. Can you possibly check what kind of cable I sent with the old device I sent you? Does the USB A plug have an white or purple inside?

**Comment by worldonfire** [comment]
> @worldonfire I'll send you an old style AAWireless cable. Can you possibly check what kind of cable I sent with the old device I sent you? Does the USB A plug have an white or purple inside?

It has white inside. I'm not 100% sure if i used that cable or not. But can give a shot when going to a ride :)

**Comment by Snirpo** [comment]
I think I included that one on purpose. The plastics of the USB A side is square and short vs the new cable with has a round-ish and longer USB A plug. Please try indeed.

**Comment by worldonfire** [comment]
Have tested this cable for some days now. Have driven ~300km (1x100km, 1x50km, 1x 70km and some shorter ones). To my surprise, it has been quite stable. Only today, after this ~50km trip, just when managed to park my car, connection was lost. But looks like, it has worked much better than the others. At least for now. Will continue testing.

As i'm testing this cable atm, gave the "hacky" one to my wife. She has used it for few longer trips now and at the moment it has been stable in her car also (other passat with harman MIB 9.2").

**Comment by Snirpo** [comment]
@worldonfire Can you also test this old style cable on the newer device?

**Comment by worldonfire** [comment]
I'm actually using my own newer device with this older cable.

**Comment by Snirpo** [comment]
@karlmorris @domik82 @xsichtasdf @Sasha0088 I still have some old style cables laying around, I can send you a cable if you want. Just pass me your address on chiel@cpeb.it.

**Comment by worldonfire** [comment]
Another week has passed and I would say that this "old style" cable is more stable. It still has some very rare drops, but not as much as the other ones (but still not as stable as the hacky one).
Would say that this is so weird. Some shorter cables don't work, then this hacky one that has propably missing shielding etc works perfect. 
@Snirpo have you had any response from the factory about these cables?

In the meantime, my wife has used the hacky one and says that it has also worked as expected (oneplus 6 + vw passat with discover pro 9.2")

Today received two different Y cables. Will test those also at some point.

**Comment by xsichtasdf** [comment]
> @karlmorris @domik82 @xsichtasdf @Sasha0088 I still have some old style cables laying around, I can send you a cable if you want. Just pass me your address on [chiel@cpeb.it](mailto:chiel@cpeb.it).

Send you my address. Let's give it a try

**Comment by karlmorris** [comment]
> > @karlmorris @domik82 @xsichtasdf @Sasha0088 I still have some old style cables laying around, I can send you a cable if you want. Just pass me your address on [chiel@cpeb.it](mailto:chiel@cpeb.it).
> 
> Send you my address. Let's give it a try 

I already had a refund processed unfortunately.

**Comment by speedyg0nz** [comment]
Owner of a Indiegogo launch unit here. Never managed to get it working over the past year with my 2018 VW Sharan running a Technisat HU, regardless of the FW update level. Read through the entire thread but it seems that there's no clear solution, other than getting a better cable? I've tried 4 different cables so far and still have no luck. If there's a specific cable to try that is known to work with Technisat HU, I'll be happy to try. That said, is the "old style cable" mentioned above the one that came with the launch unit? If so, it doesn't work as well. 

Device name (AAWireless/AndroidAuto-AAWxxxxxxx) (please complete the following information):
- AAWireless-aLScWcr6 with 2.1.0

Car (please complete the following information):
- VW Sharan 2018

Headunit (please complete the following information):
- Device part number: 5K7035200L
- Hardware: H41
- Software: 0478

Smartphone (please complete the following information):
- Sony Xperia 1 iii
- Android 12

**Comment by worldonfire** [comment]
@Snirpo looks like Tiguan with technisat has also issue with cables.
Aprox a week ago did fresh setup + used this Y cable: https://www.aliexpress.com/item/1005002704214182.html?spm=a2g0o.order_list.0.0.51c71802UUn9bT
And now looks like it has no more random disconnections.

**Comment by karlmorris** [comment]
This information may or may not be useful, but I switched to the Motorola MA1 a few weeks ago, and the dropped connection behavior is still being observed. In fact it's uncanny how similar the problems are across both units. For me this suggests the problem is in fact with the cable and/or head unit, and I've pretty much ruled out the AAWireless (and MA1) unit as being the culprit.

**Comment by Snirpo** [comment]
@karlmorris it is useful actually, because it means we are not doing anything completely stupid.
@worldonfire Definitely thanks for the info. We didn't really have any useful info from the factory yet. Still looking into this issue. What we are going to do is switch back to the old cable for our next batch. I also have some other cables coming in for a test.

**Comment by worldonfire** [comment]
This "old style" cable has been quite ok. When driving, it has been stable, but it has had few drops when starting the car or parking the car. I don't know if it is some coincidence... But it is definately usable atm.

If needed, I can help to test different cables :)

My wifes car has usb hub and my car does not so propably I can cover both B8 passat usb options (one usb port in front and second one in armrest VS only one usb in armrest). There may be some variety of usb hubs (different hw) but i'm not quite sure. Tiguan does not have usb hub.

**Comment by Snirpo** [comment]
@worldonfire I'll definitely send you some cables.

**Comment by Snirpo** [comment]
@worldonfire Getting the cables tomorrow, 20 some different cables. I'll pick out some ,test them, and forward them to you.

@xsichtasdf You should have received the old style cable on 30/8, can you confirm?

**Comment by worldonfire** [comment]
Waiting to test! :)

Got information that one tiguan in estonia is using AAWireless with such 5cm cable: https://www.aliexpress.com/item/1005001952119418.html?spm=a2g0o.order_list.0.0.554418029TY5w3
Has had very few drops.

**Comment by xsichtasdf** [comment]
> @worldonfire Getting the cables tomorrow, 20 some different cables. I'll pick out some ,test them, and forward them to you.
> 
> @xsichtasdf You should have received the old style cable on 30/8, can you confirm?

Hey, yes received. Was unable to test yet, try to do it asap!

**Comment by Snirpo** [comment]
@worldonfire got a bunch of cables now, will forward them all to you. 2 of them don't charge my Pixel 6. The cable we ship now doesn't charge it either. Not sure if that's a good indicator for a "good" or "bad" cable, but who knows. I'll tag them.

**Comment by Snirpo** [comment]
@worldonfire They are on your way

**Comment by Snirpo** [comment]
@worldonfire 

These ones don't charge my Pixel 6. Not sure if that's a good indicator, but I expect them to also be crappy in the Passat.

![PXL_20220905_055154578](https://user-images.githubusercontent.com/4076368/188479336-644f4cbf-a4c0-4b84-8c6d-17173d154d16.jpg)

These ones
![PXL_20220905_055216188](https://user-images.githubusercontent.com/4076368/188479523-3ced9e00-7094-4766-8c46-3d1e5ab9a05c.jpg)

And 

![PXL_20220905_055210785](https://user-images.githubusercontent.com/4076368/188479570-f16c0046-0d35-4115-8314-936e25b16b02.jpg)

Are the ones we prefer. There's a a good chance the red one with the metal bit will be the new cable for AAWireless.

**Comment by xsichtasdf** [comment]
Tested cable, no change of behaviour. It works for like 20 sec and than that's it. So, I'll request a refund. Thanks a lot for all your efforts, but I guess shitty VW software denies the freedom aawireless offers. :( But nonetheless, incredibly good customer support. Thanks!

**Comment by GPSMapper** [comment]
For the sake of keeping all in one place - joining here to report my issues with MIB2 STD (Discover Media 6/5")
I am facing similar disconnects, everything described in details here:
https://forum.xda-developers.com/t/aawireless-general-discussion-and-support.4247677/page-229#post-87483675

**+ Update**
My device is not rebooting but rather randomly loosing the connection with HU.
The LED behaviour is well described here: https://github.com/cpebit/aawireless-issues/issues/124#issuecomment-1156056216
so HU disconnects, led stays solid blue for another several seconds, blinking green, blue and connection is resumed. No solid green in between (do it's just reconnection, not a complete reboot)

But yes, this could happen in standstill and with mobile data turned off (to isolate Xiaomi bug which is a separate issue)

Anyway, do you need any tests or anything I can do with investigations? My device is MIB2 Discover media 6.5" (MY16) (not pro) .

VW: MIBSTD2PQNAV​
Technisat P/N: 2256110000512​
mib2_std_pq-33d305de6ec4a38d0f5b6ed693ff2635.png
HU info:
Installed SW train version: MST2_EU_VW_PQ_P0361T
SW MU version: 0361
SW version: H28.28.133_STD2NavPQ_EU
Device part number: 5C0035680D
Hardware: H32
Software: 0361
SHIELD: 2016-03 (build 020)
Media Codec: 3101.0.0.12
BT version: C194
App-Connect: 3101.358.3.1
USB Surfstick version: 3101.352.0.0

**AAW device id: AAWbbf5fd31**

I am more than ready to help as much as I can prior to giving up and sending the device back.I

**UPDATE:** I have found that there is newer update available for my HU
MST2_EU_VW_PQ_P0369T, will try to update and see if this will help

If it's better to log this as a separate GH ticket (while my device is NOT Pro/Harman, but a Technisat one) - please let me know or move yourself.

**Comment by Snirpo** [comment]
I'm debugging this issue together with @xsichtasdf at the moment. I sent him one of my MA1's as sort of a sanity check, and it does not seem to have this issue. It's highly likely this problem is hardware related and we can only fix with a new hardware revision. Luckily it does not seem to be a very widespread issue. It doesn't really seem to be related to one type of HU. It's something else in these cars, possibly related to wiring/USB harness is my guess. But that's what it still is at this point, just a guess. For @wo

**Comment by GPSMapper** [comment]
On my side I will try with SW update first and will also order a cable recommended by @worldonfire 
https://pl.aliexpress.com/item/1005002704214182.html?spm=a2g0o.order_list.0.0.51c71802UUn9bT&gatewayAdapt=glo2pol

**Comment by GPSMapper** [comment]
OK, So I have updated my HU SW to the latest release available (MST2_EU_VW_PQ_P0369T) but this has not helped.
With mobile data turned off I am still experiencing random disconnects. AAW box is definitely not rebooting  but dropping connection  - connection returns back within 5 seconds or so. No sure though where the disconnect is happening - AAW<>HU or AAW<>smartphone

Recommended Vention Y cable will arrive in 15 days, I will make a final test prior to taking a decision to return. I do not have another car in family so if it will not work in mine, it will become useless. 

For Xiaomi bug I have a workaround to test meanwhile - using an external  mobile hotspot in the car and client mode on AAW box - this should probably help to isolate Xiaomi  bug while mobile data will not be used.

Pity as I have wireless charger and wanted to go fully wireless. While the box is connected - experience is great.

**Comment by GPSMapper** [comment]
I have made another test with 2 other phones:
1. Samsung A41 / Android 12
2. Xiaomi MI6 / Android 9

with both the random HU disconnects issue is in place. So that means the issue in my case is not phone dependent.

Having this said, do you think this could be one of:
1. Faulty AAW box?
2. some bug in latest 2.1.0 AAW FW (O have seen a user on XDA complaining about disconnects started to happen on latest FW, should I try 2.0.1?

**Comment by Snirpo** [comment]
Kinda what I expected. I enabled 2.0.2 for a test. Could be faulty as well I guess, we could replace it.

**Comment by GPSMapper** [comment]
@Snirpo So, downgraded to 2.0.2 and did more tests.... More or less the same - random disconnects, BUT the box behavior is different.
2.1.0: connection loss > solid blue (2-3 sec) >  blinking green 1-2 s > blue - connect
2.0.2"connection loss > solid blue (~8 sec) >  solid green (7-8 sec) > blinking green 1-2 s > blue - connect

So it seems that with 2.0.2 FW box reboots after connection drop while with 2.1.0 with same HW and setup it was just reconnecting.
Dunno if this helps

So, while I'm waiting for the Y cable to arrive, could we agree to try box replacement?
But it would be great if you could send something grom the lab - so well tested and proved to be working stable (just to isolate all other issues). It should ne sealed - could be well used :) If it will help  - great. If not - well, VAG HU  owners are having bad luck (at least some of them).

p.s. I am in Poland and can send the device back any time.

**Comment by Snirpo** [comment]
@GPSMapper What I can do is send you a v1 device. However, I already tried that with I think @worldonfire (not sure anymore). You can pass me your address details in the ticket or on Slack https://join.slack.com/t/aawireless/shared_invite/zt-1fj6fe1it-3eX6VggGkD41CsC~PbK1zA

**Comment by worldonfire** [comment]
V1 did not make any difference. At the moment seems that only some very short cables or Y cables work. Old style original cable is somewhat stable but the hacky one was still much better...

**Comment by Snirpo** [comment]
We're still working on this issue, but it's honestly only getting more confusing...

@xsichtasdf (Technisat)
AAW V2 --> disconnects
AAW V1 --> disconnects
MA1 --> works fine

@GPSMapper (Technisat)
AAW V2 --> disconnects
AAW V1 --> works perfectly
AAW V2 replacement --> waiting for results

@worldonfire (Harman) (same as on my desk)
AAW V2--> unstable
AAW V1 --> unstable
Very short / hacked up cable --> all above stable?

**Comment by Sasha0088** [comment]
> from my side:
> 
> **2x Harman** - hacky cable works very good (it is very short, extra power not connected). Short v1 orig cable - mostly works. Have not tried Y cable yet, because gave it to friend with technisat. No matter what version of AAW, as long as settings correct (dongle mode, correct country and channel).
> 
> **1x technisat** - have not tested with very short cable/hacky cable, but with Y cable, works flawlessly. I think it is V1 device. Can verify if necessary. With original cable or some other longer cables, was unusable.
> 
> Have tested bunch of ~0.5m - 1.5m cables with my harman - none of these are stable enough to use. Ordered one 5cm cable. Will let you know, how that one will work.

Tell me if any cable helps ???

**Comment by victory123** [comment]
Hi, 
I purchases recently the AAWireless device myself, and I am using it on a VW Sharan from the year 2016.

It seems that the problems I have are related to the issues described here.

I couldn't get the device to work adequately with the supplied original short cable, as I was getting disconnections every 2 or 3 minutes. Since I changed the cable with a 1.5m long USB-A to USB-C, which I had in use before with Android Auto, the disconnections are very rare now. I noticed, that my cable sits much better in the connection port as the original one.

Is the issue still being analyzed and can we expect a solution for it?
Best regards
Victor

ID: AAWb15adba0
Version: 2.2.0

**Comment by Snirpo** [comment]
Yes, the problem has been found with a lot of help of @xsichtasdf and @worldonfire . Next production run will contain a hardware fix for this issue. The problem mainly manifest on VW and Genesis cars. Please contact support@aawireless.io, we'll replace your device with a one from the new batch when it's ready.

**Comment by lukasmrtvy** [comment]
I have the same problem, VW Tiguan 2018 ( EU ), the device keeps resetting.

**Comment by mjenny** [comment]
@Snirpo I have a Seat Alhambra with the same issue. The VW Sharan and the Seat Alhambra are almost identical with both of them using a Technisat HU.

Is there any chance that the new hardware revision will solve this also for me?

My AAWireless name would be: AAW3cdOec96 (I think it's an O and not a 0)

Thanks in advance!

**Comment by Snirpo** [comment]
Likely. We first thought it's a Harman only issue, but same thing can also happen on Technisat. It doesn't happen on all VW's though. We don't know exactly which models from which years are exactly affected. The Passat B8 model is definitely an effected model, but also not all of them are affected. This made this whole issue very confusing.

**Comment by mjenny** [comment]
Ok thanks, for the quick response. Shall I contact support@aawireless.io to be put onto the list for a replacement?

**Comment by Snirpo** [comment]
Yes

**Comment by domik82** [comment]
> Likely. We first thought it's a Harman only issue, but same thing can also happen on Technisat. It doesn't happen on all VW's though. We don't know exactly which models from which years are exactly affected. The Passat B8 is definitely an effected model, but also not all of them are affected. This made this whole issue very confusing.

Can you share what was the issue?

I'm one of the affected ones using it in VW B8. I already contacted suport and waiting for new device but I wonder what was the discovery ;) is it some electrical issue on the VW or just not handling communication properly. 

Btw do you have info when next batch of devices will be avaliable?

**Comment by TerAnYu** [comment]
I haven't used it since I received the device.
Also does not work on WV Tiguan 2015(2016) (1G) Facelift when using Discover Media (Mib2).
It is highly likely that my device will not be replaced, since many countries are against Russia.
Is it possible to fully describe what has been fixed at the hardware level and how to change it on existing devices?

**Comment by Snirpo** [comment]
It's an electrical issue indeed. Two diodes on the USB data lines which were not specced correctly. For 99% percent of the cars / head units no issue, but apparently some VW models have noisy USB data lines. This causes data corruption. Still not really an issue, because the USB protocol does error correction. But at some point it's just too much.

We cannot ship anymore to Russia indeed. Nothing personal, but that's unfortunately how the world is right now.

**Comment by TerAnYu** [comment]
https://disk.yandex.ru/i/iEvDCd08uXhPhg
https://disk.yandex.ru/i/eKoJg7jYGBuofA

Please indicate which photo shows the correct diodes and what to replace them with?
Thank you, I hope that at least this information can be passed on.

**Comment by Snirpo** [comment]
The ones from photo one. You can also just remove them. Will also do the trick. They are meant for ESD protection.

**Comment by SkyNaras** [comment]
> The ones from photo one. You can also just remove them. Will also do the trick. They are meant for ESD protection.

I have the same problem with MIB2. Can I also remove the diodes? Can you show me which ones?
![339094384_247699521007398_8327023387583217209_n](https://user-images.githubusercontent.com/10391146/231094474-fced18df-94fe-490c-8eb1-1299c591d707.jpg)

**Comment by SkyNaras** [comment]
Could you change it for me? Since the beginning, I have been practically unable to use it because it reloads regularly.
Just a plate would be enough for me, I don't need anything else.

**Comment by jaszcz0mp** [comment]
There is no way to fix this with a soldering iron on version 1.1? Unfortunately, I have the same version as @SkyNaras...

**Comment by karlmorris** [comment]
> It's an electrical issue indeed. Two diodes on the USB data lines which were not specced correctly. For 99% percent of the cars / head units no issue, but apparently some VW models have noisy USB data lines. This causes data corruption. Still not really an issue, because the USB protocol does error correction. But at some point it's just too much.
> 
> We cannot ship anymore to Russia indeed. Nothing personal, but that's unfortunately how the world is right now.

Hey, Snirpo.
I reached out to support@aawireless.io (mentioned my GitHub name) to try and procure one of the newer boards for use with my Sony XAV-AX5000 head unit. Is that email sufficient, or is there another way to get on the list?

**Comment by Sasha0088** [comment]
> 

Please write if the removal helped. Thank you 🤝🤝🤝🤝

**Comment by Snirpo** [comment]
> > It's an electrical issue indeed. Two diodes on the USB data lines which were not specced correctly. For 99% percent of the cars / head units no issue, but apparently some VW models have noisy USB data lines. This causes data corruption. Still not really an issue, because the USB protocol does error correction. But at some point it's just too much.
> > We cannot ship anymore to Russia indeed. Nothing personal, but that's unfortunately how the world is right now.
> 
> Hey @Snirpo I reached out to [support@aawireless.io](mailto:support@aawireless.io) (mentioned my GitHub name) to try and procure one of the newer boards for use with my Sony XAV-AX5000 head unit. Is that email sufficient, or is there another way to get on the list?

This is sufficient, but I'm not sure if it's going to help on a Sony. I have multiple Sony head units for testing and on these head units it's not an issue. I don't have your specific model though.

**Comment by karlmorris** [comment]
> > > It's an electrical issue indeed. Two diodes on the USB data lines which were not specced correctly. For 99% percent of the cars / head units no issue, but apparently some VW models have noisy USB data lines. This causes data corruption. Still not really an issue, because the USB protocol does error correction. But at some point it's just too much.
> > > We cannot ship anymore to Russia indeed. Nothing personal, but that's unfortunately how the world is right now.
> > 
> > 
> > Hey @Snirpo I reached out to [support@aawireless.io](mailto:support@aawireless.io) (mentioned my GitHub name) to try and procure one of the newer boards for use with my Sony XAV-AX5000 head unit. Is that email sufficient, or is there another way to get on the list?
> 
> This is sufficient, but I'm not sure if it's going to help on a Sony. I have multiple Sony head units for testing and on these head units it's not an issue. I don't have your specific model though.

Thanks, but I'm still inclined to try it. Worst case I'm left exactly where I am now.

**Comment by TerAnYu** [comment]
@Sasha0088
I apologize for the silence.
I will be able to check only after 2 weeks, when I will be discharged from the hospital.
There was a planned operation, but I hoped to have time to check it, but it did not work out.

**Comment by varghexum** [comment]
I have a very similar issue on my Kia e-Niro 2021 with random disconnects & reboots.

Can I get a new dongle when it's available?

**Comment by Snirpo** [comment]
@varghexum Probably not the same issue. AAWireless has been developed from the beginning on an e-Niro. Please contact support@aawireless for support.

@jaszcz0mp Yes, please contact support@aawireless.io

**Comment by lukasmrtvy** [comment]
Its been 14 days since I sent email.. still without any response from [support@aawireless.io](mailto:support@aawireless.io)  Has anyone similar experience?

**Comment by Snirpo** [comment]
That's not normal, I'll check.

**Comment by TerAnYu** [comment]
@Sasha0088 
today it turned out to be checked and yes, the removal of the diodes returned the functionality.

сегодня получилось проверить и да, удаление диодов вернуло работоспособность.

**Comment by Sasha0088** [comment]
Now it works flawlessly??? It is necessary to solder two resistors as in
the first photo ????

пт, 21 апр. 2023 г., 15:46 Teremshonok Anton ***@***.***>:

**Comment by TerAnYu** [comment]
es, it worked right away.
For at least 30 minutes while I was in the car, the device worked without problems.
This is wrong, but I just unsoldered the indicated diodes, although they needed to be replaced with others, but I don’t know which ones.
![2023-04-09 22-58-04](https://user-images.githubusercontent.com/7056211/233785114-89917b09-d386-4ba1-97e6-a995384d7e74.JPG)

**Comment by Snirpo** [comment]
The diode fix devices have been produced (finally). We are sending them out shortly.

**Comment by TerAnYu** [comment]
@Snirpo

Hello!
I ask you to make a small guide, if possible, which diodes are now and which ones to change, also with pictures of what to change.
I think it will be very helpful.

**Comment by lukasmrtvy** [comment]
Hey,
I got the new device, and the connection seems to be stable, but unit can not reconnect anymore after the engine shutdown, I am 
doing restore every time. Ideas?

**Comment by Snirpo** [comment]
@lukasmrtvy What do you exactly mean by restoring?

**Comment by lukasmrtvy** [comment]
After engine shutdown, the device is blinking ( green ) , but bluetooth, and/or wifi is not connected ( even after 30 mins ) on my phone.

**Comment by Snirpo** [comment]
Ok, but what do you do to "restore" the device?

**Comment by mjenny** [comment]
@Snirpo are all devices already sent out? Are we supposed to receive tracking information?

I assume it will take longer here in Switzerland due to the fact we're not in the EU.

**Comment by Snirpo** [comment]
> @Snirpo are all devices already sent out? Are we supposed to receive tracking information?
> 
> I assume it will take longer here in Switzerland due to the fact we're not in the EU.

Yes, from everyone we had the address details we sent the devices out. Please contact support for a tracking number at support@aawireless.io .

**Comment by mjenny** [comment]
> > @Snirpo are all devices already sent out? Are we supposed to receive tracking information?
> > I assume it will take longer here in Switzerland due to the fact we're not in the EU.
> 
> Yes, from everyone we had the address details we sent the devices out. Please contact support for a tracking number at [support@aawireless.io](mailto:support@aawireless.io) .

I contacted support on Sunday by replying to the original support request but so far didn't hear back from them nor did I receive the new device. Is this normal?

**Comment by Sasha0088** [comment]
I have the same situation, I also wrote to support all the data I wrote to send the device and so far no one has contacted.

**Comment by Snirpo** [comment]
No, it's not. I see the ticket, asked my guys to respond asap.

**Comment by Snirpo** [comment]
@lukasmrtvy What's the name of the device?

**Comment by jaszcz0mp** [comment]
I received a new device that works better, but it's not perfect. On the cable included in the set, it drops the connection sometimes every 20 minutes, sometimes every 5 minutes. On an adapter similar to the one in the photo, the problem occurred once every 40 hours of use.
![2023-05-11 11_34_20-Tongdaytech OTG typ C Adapter USB C męski na USB żeński USB C konwerter dla Macb](https://github.com/cpebit/aawireless-issues/assets/55209020/48e9e8f0-23d4-4852-84b0-732db9aea4b2)

**Comment by Snirpo** [comment]
@jaszcz0mp Can you give me a photo of the sticker on the device?

**Comment by jaszcz0mp** [comment]
> @jaszcz0mp Can you give me a photo of the sticker on the device?
Here you go 
![20230511_183414](https://github.com/cpebit/aawireless-issues/assets/55209020/6ccb5c68-493a-476d-a8c2-6ed76370b8b3)

**Comment by Snirpo** [comment]
@jaszcz0mp Apparently your device was accidentally sent out from our warehouse instead of the factory. I'm confirming it now, but pretty sure it's not a diode fix device. We will make sure you get a diode fix device. Sorry for the mistake.

@lukasmrtvy Your device was shipped out directly from the factory, so it should be a diode fix device. But to be sure, can you also take a photo of the sticker on the device?

**Comment by varghexum** [comment]
@Snirpo @mjenny could you please tell if this device has a diode fix?
![20230511_200749](https://github.com/cpebit/aawireless-issues/assets/51486160/a49679e3-592a-40be-96cd-e3d920238f46)

**Comment by lukasmrtvy** [comment]
![20230511_201233.jpg](https://github.com/cpebit/aawireless-issues/assets/768572/dc13e6be-08c7-4996-aeda-5092082a8a7e)

**Comment by mjenny** [comment]
@Snirpo 
Not sure if I received the correct one:
![PXL_20230513_080207149](https://github.com/cpebit/aawireless-issues/assets/4191545/f2a73900-d715-4331-a07b-c345c9f0e6fd)

**Comment by Snirpo** [comment]
@varghexum that's definitely a diode fix device.

For the other devices I asked the factory to verify, because they have quite low production numbers. I'm waiting for an answer.

**Comment by mjenny** [comment]
@Snirpo 
I know, it's only Monday but I'm still curious, did you already hear back from the factory=

**Comment by borconi** [comment]
Bit more complex.... :) too many serial numbers per device... should be solved shortly so our backend syncs with factory backend and we can handle all this queries directly...

**Comment by Snirpo** [comment]
We verified with the factory, all devices posted here are in fact diode fix devices. This fix has mainly been tested on Harman VW HU's, where it definitely fixed this issue for multiple people, including myself. 

Technisat HU's like @jaszcz0mp has, might be different story. @xsichtasdf also has a Technisat HU and is experiencing the same kind of disconnections, like @jaszcz0mp, once in half an hour or so, with a diode fix device. However, he also has an MA1, and it also drops the connection, but less. The MA1 has a short cable though. That seems to correspond with @jaszcz0mp finding of using very short cable / adapter and having almost no drops anymore. So I'm not sure if this is something we can fix.
